src/snowcli/cli/streamlit/manager.py

```diff
@@ -65,50 +37,57 @@ def share(self, streamlit_name: str, to_role: str) -> SnowflakeCursor:
     def drop(self, streamlit_name: str) -> SnowflakeCursor:
         return self._execute_query(f"drop streamlit {streamlit_name}")
 
+    def get_url_from_name(self, streamlit_name: str):
+        return self._execute_query(
+            f"call SYSTEM$GENERATE_STREAMLIT_URL_FROM_NAME('{streamlit_name}')"
+        ).fetchone()[0]
+
     def deploy(
         self,
         streamlit_name: str,
-        file: Path,
-        open_in_browser: bool,
-        use_packaging_workaround: bool,
-        packaging_workaround_includes_content: bool,
-        pypi_download: str,
-        check_anaconda_for_pypi_deps: bool,
-        package_native_libraries: str,
-        excluded_anaconda_deps: str,
+        main_file: Path,
+        environment_file: Optional[Path] = None,
+        pages_dir: Optional[Path] = None,
+        stage_name: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        replace: Optional[bool] = False,
     ):
         stage_manager = StageManager()
 
-        # THIS WORKAROUND HAS NOT BEEN TESTED WITH THE NEW STREAMLIT SYNTAX
-        if use_packaging_workaround:
-            self._packaging_workaround(
-                streamlit_name,
-                file,
-                packaging_workaround_includes_content,
-                pypi_download,
-                check_anaconda_for_pypi_deps,
-                package_native_libraries,
-                excluded_anaconda_deps,
-                stage_manager,
-            )
+        stage_name = stage_name or "streamlit"
+        stage_name = stage_manager.to_fully_qualified_name(stage_name)
 
-        qualified_name = self.qualified_name(streamlit_name)
-        streamlit_stage_name = f"snow://streamlit/{qualified_name}/default_checkout"
-        stage_manager.put(str(file), streamlit_stage_name, 4, True)
-        query_result = self._execute_query(
-            f"call SYSTEM$GENERATE_STREAMLIT_URL_FROM_NAME('{streamlit_name}')"
+        stage_manager.create(stage_name=stage_name)
+
+        root_location = stage_manager.get_standard_stage_name(
+            f"{stage_name}/{streamlit_name}"
         )
-        base_url = query_result.fetchone()[0]
-        url = self._get_url(base_url, qualified_name)
 
-        if open_in_browser:
-            typer.launch(url)
-        else:
-            return url
+        stage_manager.put(main_file, root_location, 4, True)
+
+        if environment_file and environment_file.exists():
+            stage_manager.put(environment_file, root_location, 4, True)
+
+        if pages_dir and pages_dir.exists():
```

Comment:
What if `pages_dir` is not None but does not exists?

Reply:
Good point I will add validation